src/contracts/core/AVSDirectory.sol

```diff
@@ -261,6 +271,40 @@ contract AVSDirectory is
         _deregisterFromOperatorSets(msg.sender, operator, operatorSetIds);
     }
 
+    /**
+     * @notice Called by operators to set their allocation delay.
+     * @param delay the allocation delay in seconds
+     * @dev msg.sender is assumed to be the operator
+     * @dev Fails if setting a delay would result in the ability to set an
+     *      an allocation that is active BEFORE the latest set allocation.
+     */
+    function setAllocationDelay(
+        uint32 delay
+    ) external {
+        require(
+            delegation.isOperator(msg.sender),
+            "avsDirectory.initializeAllocationDelay: operator not registered to EigenLayer yet"
+        );
+
+        AllocationDelayInfo storage delayInfo = _allocationDelayInfo[msg.sender];
+
+        require(
+            delay + block.timestamp >= delayInfo.latestAllocationEffectTimestamp,
```

Comment:
We don't need to check for `latestAllocationEffectTimestamp` and can remove it from the struct entirely